### ISSUE TITLE
Firewalld service state after watchmaker run with --no-reboot option

### ISSUE THREAD

**Issue by ewierschke** [plus3it/watchmaker#380]
Tested on CentOS 7.3/7.4 installed from the minimal ISO.  Prior to version 0.6.3 of watchmaker, use of the -n option to prevent immediate reboot after a watchmaker run, the firewalld service was left in a running state so that scripted firewall-cmd executions would succeed.  Since 0.6.3, at the end of a watchmaker run with the -n option, prior to reboot the firewalld service is left in a inactive/dead state.  This has required me to modify some install scripts, but didn't think this behavior was watchmaker's intent.

**Comment by dvu70** [comment]
Something introduced in 0.6.3 that's making firewalld inactive during the wam run? Maybe this? https://github.com/plus3it/ash-linux-formula/pull/242/commits/6ee999a7978d3f8e00885dae9564dc76ec80d8f0
Since firewalld is inactive, salt returns false for the Splunk and McAfee rules (firewalld.add_port and firewalld.reload_rules).

**Comment by lorengordon** [comment]
It's definitely ash-linux, not really watchmaker. But it's fine to track it here since in watchmaker we basically vendor a specific git reference from each formula (and thus distributing that git ref via a specific watchmaker version).

We actually tracked the problem to [this state](https://github.com/plus3it/ash-linux-formula/blob/master/ash-linux/el7/Miscellaneous/firewalld_safeties.sls#L28-L33). Something about its execution during the initial run puts firewalld into an inactive state. Rerunning the state flips it back to active, as well as rebooting. Didn't get any further in testing.

@dvu70 you want to take a stab at figuring out the magic invocation that fixes it?

**Comment by ferricoxide** [comment]
Alright, so, looks like the state-element is calling the systemd `ExecReload` unit-element:

~~~
# systemctl status -l firewalld
● firewalld.service - firewalld - dynamic firewall daemon
   Loaded: loaded (/usr/lib/systemd/system/firewalld.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Thu 2017-09-28 14:01:15 UTC; 3min 27s ago
     Docs: man:firewalld(1)
  Process: 18959 ExecReload=/bin/kill -HUP $MAINPID (code=exited, status=0/SUCCESS)
 Main PID: 1042 (code=exited, status=0/SUCCESS)

Sep 28 13:15:37 localhost systemd[1]: Starting firewalld - dynamic firewall daemon...
Sep 28 13:15:39 localhost systemd[1]: Started firewalld - dynamic firewall daemon.
Sep 28 14:01:15 ip-172-31-2-156 systemd[1]: Reloaded firewalld - dynamic firewall daemon.
~~~

And _that's_ failing. Normally `HUP` is the signal to a process to reread its configuration rather than straight-up stop. Interestingly, if one does a `systemctl reload firewalld` against a running firewalld process, it functions per normal. Not sure why the difference in behavior when executed through watchmaker. At any rate, probably need to change the logic so that it calls `try-restart` rather than `reload`.

**Comment by ferricoxide** [comment]
Looks like EL7's version of systemd/firewalld doesn't process the `reload` directive properly. The `HUP` signal _should_ result in a configuration-reread. However, the process seems to die rather than reload its config (mebbe firewalld actually wants a `SIGUSR` signal, instead?). Switching to a `try-restart` (as implemented in ferricoxide/ash-linux-formula@fcc5a65fbaab5e961b7ce2469f40a924fd615cd5) or even a generic `restart` (proposed but deemed potentially not appropriate for a "safety" state) appears to fix issue.

**Comment by ferricoxide** [comment]
"Spiffed" the method (plus3it/ash-linux-formula/pull/243). Still kludgey, but appears to work as designed both from a "watchmaker run at build-time" and retroactive-remediation. Tested converting a system's default zone from `public` to `drop` and system stayed on the air (via SSH).

Probably wants third-party validation before accepting the PR.

**Comment by ferricoxide** [comment]
Closed by plus3it/ash-linux-formula/pull/243.
